pcdsdevices/lodcm.py

```diff
@@ -60,6 +70,763 @@ def __init__(self, prefix, *args, **kwargs):
+        reflection : str or tuple
+        """
+        reflection = None
+        if self.is_diamond():
+            reflection = self.diamond_reflection.get()
+        elif self.is_silicon():
+            reflection = self.silicon_reflection.get()
+
+        if check and reflection is None:
+            raise ValueError('Unable to determine the crystal reflection')
+        if not as_tuple and reflection is not None:
+            return ''.join(map(str, reflection))
+        return tuple(reflection)
+
+    def get_material(self):
+        """
+        Get the current material.
+
+        Returns
+        -------
+        material : str
+            Material of the crystal.
+
+        Raises
+        ------
+        ValueError
+            When the material could not be determined or is something else
+             other than `Si` or `C`.
+        """
+        if self.is_diamond():
+            return 'C'
+        elif self.is_silicon():
+            return 'Si'
+        else:
+            raise ValueError(
+                "Unable to determine crystal material for Tower 2")
+
+
+class DiagnosticsTower(BaseInterface, Device):
+    """
+    Diagnostic Tower Motors.
+
+    Parameters
+    ----------
+    prefix : str
+        Epics base PV prefix.
+    """
+    # Located midway between T1 and T2 in the center of rotation of the device.
+    # horizontal slits
+    dh = FCpt(IMS, '{self._m_prefix}:MON:MMS:16',
+              kind='normal', doc='LOM Dia H')
+    # vertical slits
+    dv = FCpt(IMS, '{self._m_prefix}:MON:MMS:17',
+              kind='normal', doc='LOM Dia V')
+    dr = FCpt(IMS, '{self._m_prefix}:MON:MMS:19',
+              kind='normal', doc='LOM Dia Theta')
+    # filters wheel
+    df = FCpt(IMS, '{self._m_prefix}:MON:MMS:{self._df_suffix}', kind='normal',
+              doc='LOM Dia Filter Wheel')
+    # pips diode
+    dd = FCpt(IMS, '{self._m_prefix}:MON:MMS:18',
+              kind='normal', doc='LOM Dia PIPS')
+    # yag screen
+    yag_zoom = FCpt(IMS, '{self._m_prefix}:MON:CLZ:01',
+                    kind='normal', doc='LOM Zoom')
+
+    tab_component_names = True
+
+    def __init__(self, prefix, *args, **kwargs):
+        self._m_prefix = ''
+        self._df_suffix = '27'
+        if 'XPP' in prefix:
+            self._m_prefix = 'XPP'
+        elif 'XCS' in prefix:
+            self._m_prefix = 'HFX'
+            self._df_suffix = '22'
+
+        super().__init__(prefix, *args, **kwargs)
+
+
+class LODCMEnergySi(FltMvInterface, PseudoPositioner):
+    """
+    Energy calculations for the Si material.
+
+    Assume material is 'Si' without checking if the crystal's materials are
+    aligned for both towers.
+    """
+    tower1 = FCpt(CrystalTower1, '{self._prefix}', kind='normal')
+    tower2 = FCpt(CrystalTower2, '{self._prefix}', kind='normal')
+    dr = FCpt(IMS, '{self._m_prefix}:MON:MMS:19',
+              kind='normal', doc='LOM Dia Theta')
+
+    th1 = FCpt(OffsetMotor, prefix='{self._prefix}:TH1:OFF_Si',
```

Comment:
I should probably change these guys just to be consistent - from `th1` to `th1_Si` or `th1Si` eventually - they are after all the offset motors - but in the same time they are under the `LODCMEnergySi` object, so obviously they'll be the `Si` offsets